Dart/testData/generate/EqualsAndHashCode1.txt

```diff
@@ -5,18 +5,18 @@ class EqualsAndHashCode1 {
   static var s;
 
   @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) {
-      return true;
-    }
-    return other is EqualsAndHashCode1 &&
-        this.str == other.str &&
-        this.v == other.v &&
-        this.i == other.i;
-  }
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+          other is EqualsAndHashCode1 &&
+              str == other.str &&
+              v == other.v &&
+              i == other.i;
 
   @override
-  int get hashCode {
-    return str.hashCode ^ v.hashCode ^ i.hashCode;
-  }<caret>
+  int get hashCode =>
+      super.hashCode ^
```

Comment:
Isn't `other is ThisClass` is transitive as well? It is not symmetric. But, unlike `runtimeType`, it gives to analyzer type information for `other`. This is also the reason to leave both `other is ThisClass && runtimeType == other.runtimeType`. Case when behavior of this two implementations is not same - when class with fields has subclass with no extra fields. I am not sure, but think that in most cases is some kind of proxy, witch should be treated as instance of original class. Symmetricity problem in this case is solved by delegating `proxy.==(other)` to `other.==(this)`. So I think it is better to leave just `other is ThisClass`, if there is no option for now.